examples/experimental/i3s-17-and-debug/normals-utils.js

```diff
@@ -0,0 +1,59 @@
+import {Vector3} from 'math.gl';
+import {Ellipsoid} from '@math.gl/geospatial';
+
+const ADDITIONAL_SCALE = 200; // scaling to make normals visible on the map.
+
+export function generateBinaryNormalsDebugData(tile) {
+  if (
+    !tile.content ||
+    !tile.content.attributes ||
+    !tile.content.attributes.normals ||
+    !tile.content.attributes.positions
+  ) {
+    return {};
+  }
+
+  const normals = tile.content.attributes.normals.value;
+  const positions = tile.content.attributes.positions.value;
+
+  return {src: {normals, positions}, length: positions.length};
+}
+
+export function getNormalSourcePosition(index, data, normalsGap) {
+  let sourcePosition = {};
+
+  if (index % normalsGap === 0 || normalsGap === 0) {
+    const positions = data.src.positions;
+    const position = new Vector3([
+      positions[index * 3],
+      positions[index * 3 + 1],
+      positions[index * 3 + 2]
+    ]);
+    sourcePosition = Ellipsoid.WGS84.cartesianToCartographic(position);
+  }
+
+  return sourcePosition;
+}
+
+export function getNormalTargetPosition(index, data, normalsGap) {
+  let targetPosition = {};
+
+  if (index % normalsGap === 0 || normalsGap === 0) {
+    const positions = data.src.positions;
+    const normals = data.src.normals;
+
+    const position = new Vector3([
+      positions[index * 3],
+      positions[index * 3 + 1],
+      positions[index * 3 + 2]
+    ]);
+    const normal = new Vector3([
+      normals[index * 3],
+      normals[index * 3 + 1],
+      normals[index * 3 + 2]
+    ]).multiplyByScalar(ADDITIONAL_SCALE);
+    targetPosition = Ellipsoid.WGS84.cartesianToCartographic(position.add(normal));
```

Comment:
I think we should just be able to use the Ellipsoid to get a 4x4 transformation matrix. Doing the full conversion every time will be quite slow.

Reply:
Actually this functions implements [Performance optimization logic](https://deck.gl/docs/developer-guide/performance).
To display source and target points I need transform them to cartographic coordinates. It is not clear for me what transformation matrix do you mean? I didn't find any 4x4 transformation matrix in Ellipsoid module.

Reply:
https://github.com/uber-web/math.gl/blob/master/modules/geospatial/src/ellipsoid/ellipsoid.js#L123-L129